app/src/main/java/org/oppia/app/recyclerview/BindableAdapter.kt

```diff
@@ -32,17 +33,17 @@ class BindableAdapter<T : Any> internal constructor(
   private val viewHolderFactoryMap: Map<Int, ViewHolderFactory<T>>,
   private val dataClassType: KClass<T>
 ) : RecyclerView.Adapter<BindableAdapter.BindableViewHolder<T>>() {
-  private val dataList: MutableList<T> = ArrayList()
+  private var dataList: MutableList<T> = ArrayList()
```

Comment:
Could we do a shallow copy of the list & diff that, instead? Prefer minimizing mutability.

Reply:
Resolving this on behalf of ben